aioxmpp/stream.py

```diff
@@ -2655,16 +2656,23 @@ def handler_error(exc):
             listener.cancel()
             raise
 
-        if not timeout:
-            reply = yield from fut
-        else:
+        try:
+            if not timeout:
+                reply = yield from fut
+            else:
+                try:
+                    reply = yield from asyncio.wait_for(
+                        fut,
+                        timeout=timeout
+                    )
+                except asyncio.TimeoutError:
+                    raise TimeoutError
```

Comment:
Do we want `TimeoutError` instead of `asyncio.TimeoutError` beyond being backwards compatible, I know we use the system level exceptions at other places – but do we really want that.

What does a grep through the code base say? (That is a discussion that must be concluded before aioxmpp 1.0).

Reply:
I intentionally chose TimeoutError here. If you want asyncio.TimeoutError, you can implement it yourself with ``asyncio.wait_for`` or something.